filebeat/prospector/redis/prospector.go

```diff
@@ -52,17 +57,22 @@ func (p *Prospector) LoadStates(states []file.State) error {
 func (p *Prospector) Run() {
 	logp.Debug("redis", "Run redis prospector with hosts: %+v", p.config.Hosts)
 
+	if len(p.config.Hosts) == 0 {
+		return
+	}
+
+	forwarder, err := harvester.NewForwarder(p.cfg, p.outlet)
+	if err != nil {
+		logp.Err("Error: %s", err)
+		return
+	}
+
 	for _, host := range p.config.Hosts {
 		pool := CreatePool(host, p.config.Password, p.config.Network,
 			p.config.MaxConn, p.config.IdleTimeout, p.config.IdleTimeout)
 
-		var err error
-
 		h := NewHarvester(pool.Get())
-		h.forwarder, err = harvester.NewForwarder(p.cfg, p.outlet)
-		if err != nil {
-			logp.Err("Error: %s", err)
-		}
+		h.forwarder = forwarder
```

Comment:
yes, requirements on ordering in registrar did force me to make the beat.Client thread-safe (big mutex)